src/uct/sm/cma/cma_ep.c

```diff
@@ -49,8 +50,11 @@ ucs_status_t uct_cma_ep_put_zcopy(uct_ep_h tl_ep, const void *buffer, size_t len
 
     delivered = process_vm_writev(ep->remote_pid, &local_iov, 1, &remote_iov, 1, 0);
     if (ucs_unlikely(delivered != length)) {
+        if (delivered < 0) {
+            ucs_error("process_vm_writev reported an error: %m");
+        }
```

Comment:
It is consistent and clear way. First we return the error, then we return the  status.